pkg/machine/controller/machine/controller.go

```diff
@@ -83,8 +84,13 @@ func isCustom(obj *v3.Machine) bool {
 func (m *Lifecycle) Create(obj *v3.Machine) (*v3.Machine, error) {
 	if isCustom(obj) {
 		m.setupCustom(obj)
-		v3.MachineConditionReady.True(obj)
-		return obj, nil
+		newObj, err := v3.MachineConditionInitialized.Once(obj, func() (runtime.Object, error) {
+			if err := validateCustomHost(obj); err != nil {
+				return obj, errors.New("CustomSshConfig is invalid")
```

Comment:
Your masking the original error.  Better to use errors.Wrap

Reply:
@ibuildthecloud yeah, i masked for a reason. but the original error is too vague. If you give an invalid key it just says that `sshKey not found`. Let me think of a way to give a better error.